### PR TITLE
chore(EMS-3957): dry countries, currencies calls

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
Each of the following UI controllers were making 2x API calls to get countries and currencies:

- Check your answers - Export contract
- Check your answers - Policy
- Export contract - Agent charges
- Policy - Check your answers

This PR introduces a new GQL resolver so that countries and currencies can be obtained from the API with 1x call.

### API
- Create new `getCountriesAndCurrencies` GQL resolver.
- Update `ALLOWED_GRAPHQL_RESOLVERS`.
- Update `getApimCisCountries` GQL query to consume `apimCisCountries` helper.
- Update `getApimCurrencies` GQL query to consume `apimCisCurrencies` helper.
- Update `apimCisCountries` and `apimCisCurrencies` helpers to have an object structure.

### UI
- Create new `getCountriesAndCurrencies` GQL call/query.
- Update all controllers that have `api.keystone.countries.getAll` and `api.keystone.APIM.getCurrencies` to call `api.keystone.getCountriesAndCurrencies`.
- Update unit tests.

## Miscellaneous :heavy_plus_sign:
- Remove unused "page" GQL resolver.
- Add some missing return types.
- Update test mocks.
- Minor documentation improvements.

### CODE REVIEW

File: src/api/custom-resolvers/queries/get-APIM-CIS-countries/index.test.ts [hunk 1]
```diff
@@ -1,68 +1,49 @@
...
+  describe('when there is an error', () => {
     beforeEach(() => {
-      APIM.getCisCountries = mockSpyPromiseRejection;
+      apimCisCountries.get = mockSpyPromiseRejection;
     });
 
     it('should throw an error', async () => {
       try {
-        await getApimCisCountries();
+        await getApimCisCountriesQuery();
```
Comment (Zainzzkk) [inline src/api/custom-resolvers/queries/get-APIM-CIS-countries/index.test.ts:44]: could we use rejects.ToThrow( here?

File: src/api/custom-resolvers/queries/get-countries-and-currencies/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @param {Context} context: KeystoneJS context API
+ * @returns {Promise<Object>} APIM countries and currencies
+ */
+const getCountriesAndCurrencies = async (root: any, variables: object, context: Context) => {
+  try {
+    console.info('Getting countries and currencies (getCountriesAndCurrencies helper)');
+
+    const [countries, currenciesResponse] = await Promise.all([await getCountries(context), await apimCurrencies.get()]);
+
+    if (!currenciesResponse.success) {
```
Comment (Zainzzkk) [inline src/api/custom-resolvers/queries/get-countries-and-currencies/index.ts:20]: do we not need a check for if no countries as well?

Comment (ttbarnes) [reply src/api/custom-resolvers/queries/get-countries-and-currencies/index.ts:20]: No - `getCountries` is an internal async call which just calls our DB. It does not return `success: true`, it returns an array. If this fails, an error will be thrown and this new function/resolver will catch it.

The other function `apimCurrencies.get`, is an external async call which _does_ return `success: true`.

File: src/api/custom-resolvers/queries/get-APIM-CIS-countries/index.test.ts (hunk 1 above)
Comment (ttbarnes) [reply src/api/custom-resolvers/queries/get-APIM-CIS-countries/index.test.ts]: Great spot, updated

File: src/api/custom-resolvers/queries/get-countries-and-currencies/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    });
+  });
+
+  describe('when there is an error', () => {
+    beforeEach(() => {
+      apimCurrencies.get = mockSpyPromiseRejection;
+    });
+
+    it('should throw an error', async () => {
+      try {
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/api/custom-resolvers/queries/get-countries-and-currencies/index.test.ts]: Great spot, updated

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**